upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -189,6 +189,35 @@ func (tf *TemplateFunctions) GetInstanceGroup(name string) (*kops.InstanceGroup,
 	return nil, fmt.Errorf("InstanceGroup %q not found", name)
 }
 
+// CloudControllerConfigArgv returns the args to external cloud controller
+func (tf *TemplateFunctions) CloudControllerConfigArgv() ([]string, error) {
+	if tf.cluster.Spec.ExternalCloudControllerManager == nil {
+		return nil, fmt.Errorf("ExternalCloudControllerManager is nil")
+	}
+	var argv []string
+
+	if tf.cluster.Spec.ExternalCloudControllerManager.LogLevel != 0 {
+		argv = append(argv, fmt.Sprintf("--v=%d", tf.cluster.Spec.ExternalCloudControllerManager.LogLevel))
+	} else {
+		argv = append(argv, "--v=2")
+	}
+	// if tf.cluster.Spec.ExternalCloudControllerManager.LogLevel != "" {
+	// 	argv = append(argv, fmt.Sprintf("--watch-namespace=%s", tf.cluster.Spec.ExternalCloudControllerManager.LogLevel))
+	// }
+	if tf.cluster.Spec.ExternalCloudControllerManager.CloudProvider !=""{
+		argv = append(argv, fmt.Sprintf("--cloud-provider=%s",  tf.cluster.Spec.ExternalCloudControllerManager.CloudProvider))
+	}else if tf.cluster.Spec.CloudProvider != "" {
+		argv = append(argv, fmt.Sprintf("--cloud-provider=%s", tf.cluster.Spec.CloudProvider))
+	}else {
+		return nil, fmt.Errorf("Cloud Provider is not set")
+	}
+
+	if tf.cluster.Spec.ExternalCloudControllerManager.ClusterName != "" {
+		argv = append(argv, fmt.Sprintf("--cluster-name=%s", tf.cluster.Spec.ExternalCloudControllerManager.ClusterName))
```

Comment:
I do not see that it does anything in OpenStack external ccm?

Reply:
`cluster-name` will be reflected in resources names created in OpenStack like loadbalancer name for example.   I need this to run multiple k8s clusters in one OpenStack project and create loadbalancers for them.

Reply:
in my opinion that is not true, I am quite familiar with external cloudprovider code and that is not used in loadbalancer name.

https://github.com/kubernetes/cloud-provider-openstack/search?q=cluster-name&unscoped_q=cluster-name

Reply:
I use non octavia loadbalancer and naming follows this scheme: https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go#L505

Reply:
maybe that logic is then coming from some another component because ccm does not have parser for that flag at all. However, when I compile ccm there is field for that in --help

```
      --cluster-name string                                     The instance prefix for the cluster. (default "kubernetes")
```

Reply:
@zetaab The external cloud provider uses the cloud controller config from k8s cloud controller manager. And the cloud controller is started with machinery from k8s (k8s.io/kubernetes/cmd/cloud-controller-manager/app) which makes it use certain config values. I however don't know if the config values are propagated and used within the actual controller.